subsys/bluetooth/host/gatt.c

```diff
@@ -1958,7 +1988,7 @@ static void sc_restore(struct bt_conn *conn)
 	}
 
 	/* Reset config data */
-	sc_clear(cfg);
```

Comment:
It is not a bug fix, just a slight change of how SC settings are handled.

We previously removed the SC setting when restoring (sc_clear). Now we maintain the setting but reset the data to 0 when the SC indication has been sent and only remove the setting when the device unpairs or unsubscribes.

It allows us to keep track of devices that are subscribed to SC without having to loop through CCC data of all bonded devices, because we want CCC data to be loaded on demand when a device connects.